.github/workflows/tests.yml

```diff
@@ -119,24 +131,90 @@ jobs:
 
       - name: Start Covalent dispatcher server
         run: covalent start
+               
+      - name: Run SDK tests and measure coverage
+        if: ${{ steps.working-directory.outputs.sdk}} == 'true'
+        uses: nick-fields/retry@v2
+        with:
+          timeout_minutes: 15
+          max_attempts: 2
+          on_retry_command: rm -f ~/.config/covalent/covalent.conf
+          command: |
+            PYTHONPATH=$PWD/ pytest -vv tests/covalent_tests  --cov=covalent --cov-config=.coveragerc
+            
+      - name: Generate SDK coverage report
+        run: coverage xml -o sdk_coverage.xml
 
-      - name: Run tests and measure coverage
+      - name: Run dispatcher tests and measure coverage
+        if: ${{ steps.working-directory.outputs.dispatcher }} == 'true'
         uses: nick-fields/retry@v2
         with:
-            timeout_minutes: 30
-            max_attempts: 2
-            on_retry_command: rm -f ~/.config/covalent/covalent.conf
-            command: |
-              PYTHONPATH=$PWD/ pytest tests -vv --ignore=tests/qa --ignore=tests/stress_tests  --cov=covalent --cov=covalent_dispatcher 
+          timeout_minutes: 15
+          max_attempts: 2
+          on_retry_command: rm -f ~/.config/covalent/covalent.conf
+          command: |
+            PYTHONPATH=$PWD/ pytest -vv tests/covalent_dispatcher_tests  --cov=covalent_dispatcher --cov-config=.coveragerc
+      
+      - name: Generate dispatcher coverage report
+        run: coverage xml -o dispatcher_coverage.xml
```

Comment:
some questions about how all this works:

how does it know that `dispatcher_coverage.xml` refers to `--cov=covalent_dispatcher`?
how does codecov know to only look at the percentage of `covalent_dispatcher` that's covered by `covalent_dispatcher_tests`?